packages/react-core/__tests__/operations/groupby.test.js

```diff
@@ -24,125 +25,91 @@ const COLUMN = 'test';
+        { name: 'Category 1', value: 6 }
+      ]
+    ];
 
-    test(AggregationTypes.AVG, () => {
-      const groups = groupValuesByColumn(
-        buildValidFeatures(COLUMN),
-        `${COLUMN}_quantitative`,
-        `${COLUMN}_qualitative`,
-        AggregationTypes.AVG
-      );
-      expect(groups).toEqual([
-        {
-          name: 'Category 2',
-          value: 3
-        },
-        {
-          name: 'Category 1',
-          value: 3
-        }
-      ]);
+    describe('one valuesColumns', () => {
+      Object.values(AggregationTypes).forEach((operation, idx) => {
+        test(operation, () => {
+          const groups = groupValuesByColumn({
+            data: buildValidFeatures(COLUMN),
+            valuesColumns: [`${COLUMN}_quantitative`],
+            keysColumn: `${COLUMN}_qualitative`,
+            operation
+          });
+          expect(groups).toEqual(RESULTS[idx]);
+        });
+      });
     });
 
-    test(AggregationTypes.MIN, () => {
-      const groups = groupValuesByColumn(
-        buildValidFeatures(COLUMN),
-        `${COLUMN}_quantitative`,
-        `${COLUMN}_qualitative`,
-        AggregationTypes.MIN
+    describe('multiple valuesColumns', () => {
+      const RESULTS_FOR_MULTIPLE = RESULTS.map((result, idx) =>
+        result.map(({ name, value }) => ({
+          name,
+          // === 0 is AggregationTypes.COUNT
+          value: value * (idx === 0 ? 1 : 2)
+        }))
       );
-      expect(groups).toEqual([
-        {
-          name: 'Category 2',
-          value: 1
-        },
-        {
-          name: 'Category 1',
-          value: 2
-        }
-      ]);
-    });
 
-    test(AggregationTypes.MAX, () => {
-      const groups = groupValuesByColumn(
-        buildValidFeatures(COLUMN),
-        `${COLUMN}_quantitative`,
-        `${COLUMN}_qualitative`,
-        AggregationTypes.MAX
-      );
-      expect(groups).toEqual([
-        {
-          name: 'Category 2',
-          value: 5
-        },
-        {
-          name: 'Category 1',
-          value: 4
-        }
-      ]);
-    });
-
-    test(AggregationTypes.SUM, () => {
-      const groups = groupValuesByColumn(
-        buildValidFeatures(COLUMN),
-        `${COLUMN}_quantitative`,
-        `${COLUMN}_qualitative`,
-        AggregationTypes.SUM
-      );
-      expect(groups).toEqual([
-        {
-          name: 'Category 2',
-          value: 9
-        },
-        {
-          name: 'Category 1',
-          value: 6
-        }
-      ]);
+      Object.values(AggregationTypes).forEach((operation, idx) => {
+        test(operation, () => {
+          const groups = groupValuesByColumn({
+            data: buildValidFeatures(COLUMN),
+            valuesColumns: [`${COLUMN}_quantitative`, `${COLUMN}_quantitative_2`],
+            joinOperation: AggregationTypes.SUM,
+            keysColumn: `${COLUMN}_qualitative`,
+            operation
+          });
+          expect(groups).toEqual(RESULTS_FOR_MULTIPLE[idx]);
```

Comment:
This relies on `RESULTS_FOR_MULTIPLE` to be the same order as the result of `Object.values(AggregationTypes)`. 
But key "order" on objects is not strictly guaranteed, and anyway this isn't very easy to read.

Declaring `RESULTS_FOR_MULTIPLE` as an object with the types as keys would solve this.

Reply:
I like it. I'll change it. Thanks @bbecquet.

Reply:
Changed.